processor/blockprocessor/block_processor.go

```diff
@@ -0,0 +1,78 @@
+package blockprocessor
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/processor"
+)
+
+type blockProcessor struct {
+	handler            func(block *ledgercore.ValidatedBlock) error
+	nextRoundToProcess uint64
+	ledger             *ledger.Ledger
+}
+
+// MakeProcessor creates a block processor
+func MakeProcessor(ledger *ledger.Ledger, handler func(block *ledgercore.ValidatedBlock) error) processor.Processor {
+	return &blockProcessor{ledger: ledger, nextRoundToProcess: 1, handler: handler}
+}
+
+// Process a raw algod block
+func (processor *blockProcessor) Process(cert *rpcs.EncodedBlockCert) error {
+	if processor.ledger == nil {
+		return fmt.Errorf("Process() err: local ledger not initialized")
+	}
+	if cert == nil {
+		return fmt.Errorf("Process() err: cannot process a nil block")
+	}
+	if uint64(cert.Block.Round()) != processor.nextRoundToProcess {
+		return fmt.Errorf("Process() err: block has invalid round")
```

Comment:
If you want to be consistent with other indexer code, `err: ...` describes the value of the variable `err`. Here, there is no variable `err`, so writing `err: ` is not appropriate.

Instead you can have
```
return fmt.Errorf("Process() invalid round cert.Block.Round(): %d processor.nextRoundToProcess: %d", cert.Block.Round(), processor.nextRoundToProcess)
```